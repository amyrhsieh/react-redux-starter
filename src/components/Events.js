import React from 'react';
import { connect } from 'react-redux';

const Events = props => (
  <div>
    Events
    <ul>
      {props.events.map(event =>
        (<li key={event.id}>{event.name}</li>)
      )}
    </ul>
  </div>
);

const mapStateToProps = state => ({
  events: state.events
});

const VisibleEvents = connect(
  mapStateToProps
)(Events);

export default VisibleEvents;
