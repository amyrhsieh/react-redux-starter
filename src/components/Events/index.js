import React from 'react';
import { connect } from 'react-redux';

export const Events = props => (
  <div>
    <h2>Events</h2>
    <ul>
      {props.events.map(event =>
        (<li key={event.id}>{event.name}</li>)
      )}
    </ul>
  </div>
);

const mapStateToProps = state => ({
  events: state.events
});

const VisibleEvents = connect(
  mapStateToProps
)(Events);

export default VisibleEvents;
