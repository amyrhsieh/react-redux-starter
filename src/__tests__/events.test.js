import React from 'react';
import renderer from 'react-test-renderer';
import { Events } from 'Components/Events';
import events from 'Mocks/eventsMock';

describe('Events (Snapshot)', () => {
  it('Events component renders Events', () => {
    const component = renderer.create(<Events events={events} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
