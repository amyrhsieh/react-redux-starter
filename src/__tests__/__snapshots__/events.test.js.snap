// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events (Snapshot) Events component renders Events 1`] = `
<div>
  <h2>
    Events
  </h2>
  <ul>
    <li>
      Grand Opening
    </li>
    <li>
      Special sale
    </li>
  </ul>
</div>
`;
